src/main/java/mate/academy/bookstore/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,63 @@
+package mate.academy.bookstore.exception;
+
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                                    .map(this::getErrorMessage)
+                                    .toList();
+        return getResponseEntity(HttpStatus.BAD_REQUEST, errors);
+    }
+
+    @ExceptionHandler(SQLIntegrityConstraintViolationException.class)
```

Comment:
I assume that is the exception that you get when you try to save the duplicate ISBN to the DB. That's a bad practice to pass the raw SQL exception to the user, we should prevent them from happening by adding a ISBN uniqueness check to the service create/update methods, and throwing & handling some custom exception instead, where the status code would be a `400 - BAD REQUEST` because it is the user who sent us this invalid isbn

Reply:
So if I understand right i should change the status to `HttpStatus.BAD_REQUEST`?

Reply:
not really 😄 
we should add a check with some `if (repo.existsByIsbn(isbn)) {throw CustomEx}` to the create and update methods of the book service, to prevent this SQL exception from happening, and here we just replace this handle of SQL exception with a handling of our that custom exception